### PR TITLE
Fix remove error, add flags

### PR DESCRIPTION
**Pull request by Psayker**
* Add new args for detailed settings
* Fix error with remove
* Unified prefix for all driver args and environment variables.

Error with remove - if we manual delete VM from Proxmox VE, we can obtain misdeleate only by VMID

Archive with binaries, if you have any error with downloading from repo
[docker-machine-driver-proxmoxve.macos-amd64.tar.gz](https://github.com/lnxbil/docker-machine-driver-proxmox-ve/files/4120906/docker-machine-driver-proxmoxve.macos-amd64.tar.gz)
[docker-machine-driver-proxmoxve.linux-amd64.tar.gz](https://github.com/lnxbil/docker-machine-driver-proxmox-ve/files/4120907/docker-machine-driver-proxmoxve.linux-amd64.tar.gz)

### CODE REVIEW

**Review by lnxbil — COMMENTED** [review]
Just curious:
This is a nice idea, but why do we need this? Or more precisely, what problem does this solve?

**Comment by Psayker** [comment]
You have problem with `Remove` method, it delete by VMID VM in Proxmox. 
Step for reproduce this error:
1. Create via docker-machine VM
2. Delete manualy in Proxmox VM
3. Create new VM
4. run `docker-machine rm vm_name`
On stage 4, docker-machine use VMID, that saved localy, and send delete request to this VMID, but VM created by docker-machine already deleted and some new VM was created and assigned by this VMID.

New flags give us more flexible settings for our VM. Unified driver name, help when you want build custom driver or change it, and get both versions, modified and base.

File: proxmoxdriver.go [hunk 1]
```diff
@@ -90,117 +122,148 @@ func (d *Driver) connectAPI() error {
...
+			EnvVar: upperDriverName + "_VM_MEMORY",
+			Name:   driverName + "-vm-memory",
 			Usage:  "memory in GB",
 			Value:  8,
 		},
 		mcnflag.StringFlag{
-			EnvVar: "PROXMOXVE_VM_CPU",
-			Name:   "proxmoxve-vm-cpu-cores",
+			EnvVar: upperDriverName + "_VM_SCSI_ARGS",
+			Name:   driverName + "-vm-scsi-args",
```
Comment (onegreyonewhite) [inline proxmoxdriver.go:190]: Cover #28

**Comment by Psayker** [comment]
Fix few mistakes, delete hardcode for thinpool, directive if in Makefile

**Comment by lnxbil** [comment]
> You have problem with `Remove` method, it delete by VMID VM in Proxmox.
> Step for reproduce this error:
> 
> 1. Create via docker-machine VM
> 2. Delete manualy in Proxmox VM
> 3. Create new VM
> 4. run `docker-machine rm vm_name`
>    On stage 4, docker-machine use VMID, that saved localy, and send delete request to this VMID, but VM created by docker-machine already deleted and some new VM was created and assigned by this VMID.

Oh sorry, now I get what the problem is. The solution is very nice.

We're added a lot of complexity to the whole driver (in comparison to other drivers) and need a good test environment. Any ideas how to set one up?

Mine looks currently like this:
I created a VM in my PVE with nested PVE inside that has LVM, ZFS, NFS, Directory storage to test that. I was unable to install a single-node ceph setup and the env is still lacking lvm-thin and iscsi. This is only the storage side. We need also network testing (for the vlan stuff) and now also the manual deletion of a VM and possibly any other combination of flags.

**Comment by onegreyonewhite** [comment]
VLAN can be safely used in OpenVSwitch (we checked this). In VE proxy, in principle, only file-oriented storage and all the others are different, so it’s enough to add only verification with lvm-thinpool.
One way or another, it is enough to verify the correctness of the settings passed, and testing Proxmox VE functionality would be unnecessary.

**Comment by travisghansen** [comment]
@onegreyonewhite this is a great PR. I have implemented most of the features in a new branch here https://github.com/lnxbil/docker-machine-driver-proxmox-ve/pull/34 if you'd like to collaborate at all.

Essentially I've made it much more 'cloud-native' in behavior with `cloud-init` support and using `clone` functionality.

**Comment by onegreyonewhite** [comment]
@travisghansen thanks for the tip. I think we're going to fork this project later and will develop it independently, because it is unacceptable to wait six months for pull requests to be accepted.

**Comment by travisghansen** [comment]
Fair enough. I’ve pretty well got it in a final state. Just starting to test now with rancher actually so I’m sure there will be some nuances to clean up as a result of that.

**Comment by lnxbil** [comment]
> @travisghansen thanks for the tip. I think we're going to fork this project later and will develop it independently, because it is unacceptable to wait six months for pull requests to be accepted.

Sorry about that. My time for open source is very limited at the moment.

**Comment by lnxbil** [comment]
As @travisghansen said, included so I close this pull request

**Pull request closed without merging**